.pre-commit-config.yaml

```diff
@@ -150,9 +150,15 @@ repos:
             (?x)
             \#\ type:\ (?!ignore)|
             \#\ type:\s?ignore(?!\[)|
-            \)\ ->\ \"
+
+            # string return type annotation
+            \)\ ->\ \"|
+
+            # string type annotation in generic
+            (Callable|Dict|Iterable|Iterator|List|Mapping|MutableMapping|Sequence|Set|Tuple|Type|Union)\[[^\]]*\".+?\"[^\]]*\]
         language: pygrep
         types: [python]
+        exclude: ^pandas/_typing\.py$
```

Comment:
is this because of aliases? If so, we may also define aliases local to a module, so we would need to allow if preceeded by `=`

Reply:
we also have the issue of casts, which still require the quoting.

Reply:
It's not because of aliases (they're fine), but because in `_typing` we would otherwise get
```python-traceback
ImportError while loading conftest '/home/marco/pandas-marco/pandas/conftest.py'.
pandas/__init__.py:22: in <module>
    from pandas.compat.numpy import (
pandas/compat/__init__.py:14: in <module>
    from pandas._typing import F
pandas/_typing.py:79: in <module>
    FrameOrSeriesUnion = Union["DataFrame", Series]
E   NameError: name 'Series' is not defined
```
because `Series` is only defined if `TYPE_CHECKING` (same situation as in the other comment)

Reply:
yep, I wasn't clear. it's not just type aliases https://www.python.org/dev/peps/pep-0563/#forward-references

> This PEP addresses the issue of forward references in type annotations. The use of string literals will no longer be required in this case. However, there are APIs in the typing module that use other syntactic constructs of the language, and those will still require working around forward references with string literals. The list includes: